modules/transport-netty4/src/main/java/org/opensearch/http/netty4/Netty4HttpServerTransport.java

```diff
@@ -335,38 +353,152 @@ protected HttpChannelHandler(final Netty4HttpServerTransport transport, final Ht
             this.responseCreator = new Netty4HttpResponseCreator();
         }
 
+        public ChannelHandler getRequestHandler() {
+            return requestHandler;
+        }
+
         @Override
         protected void initChannel(Channel ch) throws Exception {
             Netty4HttpChannel nettyHttpChannel = new Netty4HttpChannel(ch);
             ch.attr(HTTP_CHANNEL_KEY).set(nettyHttpChannel);
             ch.pipeline().addLast("byte_buf_sizer", byteBufSizer);
             ch.pipeline().addLast("read_timeout", new ReadTimeoutHandler(transport.readTimeoutMillis, TimeUnit.MILLISECONDS));
+
+            configureDefaultPipeline(ch);
+            transport.serverAcceptedChannel(nettyHttpChannel);
+        }
+
+        @Override
+        public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+            ExceptionsHelper.maybeDieOnAnotherThread(cause);
+            super.exceptionCaught(ctx, cause);
+        }
+
+        protected void configureDefaultPipeline(Channel ch) {
```

Comment:
Nit: is there a non-default pipeline?

Reply:
Yes, the `security` plugin changes it [1], or may be the naming is confusing?

[1] https://github.com/opensearch-project/security/pull/2051/files#diff-e1d270229cf186b8d08401506104e39ec8d161bd9a54cd66c6657d0e60fa6bacR111

Reply:
I think it's the naming, I would expect it to be called `createPipeline()`, but you must have had a reason not to?